base/genimport/importer.go

```diff
@@ -261,22 +267,125 @@ func removeAllFilesInDirExcept(o *Output, dir string, except_list []string) {
 		if name == "" {
 			continue
 		}
-		filepath := paths.Subdir(dir, name)
-		if err := os.Remove(filepath); err != nil {
-			o.Errorf("error removing file %q: %v", filepath, err)
+		f := paths.Subdir(dir, name)
+		if err := os.Remove(f); err != nil {
+			o.Errorf("error removing file %q: %v", f, err)
 		}
 	}
 }
 
 func createPluginGoModFile(o *Output, pkgpath string, dir string) string {
+	file := modfile.File{}
+	err := file.AddModuleStmt("gomacro.imports/" + pkgpath)
+	if err != nil {
+		o.Errorf("error setting module in go.mod", err)
+	}
+
+	// Attempt to use the local module if present.
+	// This only works if the import shares a mod file with current working
+	// directory, because we only know to guess "." for the local location.
```

Comment:
This suggests another idea, which is that some config file or magic-statement could say "also look in these paths for local copies of modules", and then you can import those, too, with ordinary import syntax.  Maybe that's what you're getting at with `:replace`.

Reply:
It's similar, but the idea of a config file did occur to me over the weekend since this should probably also support specific versions of imported packages like a mod file, which neither the `./` nor `:replace` approaches really adequately solve. But I think all solutions are a lot more than what we need for our uses, so I'm good with punting on that for now and playing with some POCs in my free time or something.